### ISSUE TITLE
Subject Namespace

### ISSUE THREAD

**Issue by simonFuturePerfect** [ehrbase/ehrbase#1058]
We use a URL for the Subject Namespace unfortunately EHRBase doesn't like the period '.'. This prevents a namespace such as "https://fhir.nhs.uk/Id/nhs-number" from being used. Additionally this is not consistent with other CDR providers.  Is it possible to alter the Subject Namespace validation.

Thanks Simon

**Comment by simonFuturePerfect** [comment]
# Environment information

EHRBASE_VERSION: 0.22.0-SNAPSHOT
Java Runtime version: Open JDK 11.0.16
OS Version: Debian GNU/Linux 9
Archie version: 2.0.1
PostgreSQL version:11 (Azure PostgreSQL)

# Steps to reproduce

curl -X 'PUT' \
  'https://anapplication.azurewebsites.net/ehrbase/rest/openehr/v1/ehr/f56175ac-0e72-4b7b-9088-233652444371' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "_type": "EHR_STATUS",

"name": {
    "_type": "DV_TEXT",
    "value": "EHR Status"
  },
  "subject": {
    "_type": "PARTY_SELF",
  "external_ref": {
      "_type": "PARTY_REF",
      "type": "PERSON",
      "id": {
       "_type": "GENERIC_ID",
        "value": "9044258125",
        "scheme": "id_scheme"
      },
      "namespace": "https://fhir.nhs.uk.Id/nhs-number"
    }
  },
  "is_queryable": true,
  "is_modifiable": true,
 "archetype_node_id": "openEHR-EHR-EHR_STATUS.generic.v1"
}'

# Actual result
Error Status Code 400 Bad Request
{
  "error": "Bad Request",
  "message": "Subject's namespace format invalid"
}

# Expected result

200 EHrId Created

# Further information

I think the RegEx in 
**ValidationServiceImp** needs to include the '.' character so it should be
private static final Pattern NAMESPACE_PATTERN = Pattern.compile("[a-zA-Z][a-zA-Z0-9-_:/&+?.]*");

I'm happy to create a unit test that emulates the openEHR subject namespace requirements, if it helps.

**Comment by vladislavploaia** [comment]
Hello @simonFuturePerfect,
Regex expression in specifications, is the following:
[a-zA-Z][a-zA-Z0-9_.:\/&?=+-]* - https://specifications.openehr.org/releases/BASE/latest/base_types.html#_object_ref_class
Your expression is the following:
[a-zA-Z][a-zA-Z0-9-_:/&+?.]*

Question: What is the expected one? I would go with what I see in the specifications (both cases can find the URL https://fhir.nhs.uk.id/nhs-number in namespace).

Please let me know, so I can report bug for that, because currently, I see this regex [a-zA-Z][a-zA-Z0-9-_:/&+?]* in **ValidationServiceImp.java** and that one is not able to see the dot.
Old ticket was closed for unclear reasons (the same error): https://github.com/ehrbase/ehrbase/issues/399

CC: @vidi42 , @birgerhaarbrandt

**Comment by simonFuturePerfect** [comment]
I agree with you @vladislavploaia  go with the one that is in the specification. I just added the '.' to the RegExpression in **ValidationServiceImp.java**. Thanks for the help.

**Comment by vladislavploaia** [comment]
@simonFuturePerfect,
Defect ticket reported in our internal Jira: https://jira.vitagroup.ag/browse/CDR-754
Once I get any updates, I will let you know.
Thank you!
CC: @vidi42 , @stefanspiska
